### PR TITLE
net/nanocoap: document internal option functions

### PR DESCRIPTION
**Pull request by kb2ma**
### Contribution description
Adds internal documentation for a couple of functions related to option handling. I found myself needing to understand the inputs and output in detail, and this documentation means we don't have to figure it out each time. ;-)

### Testing procedure
N/A, just documentation

### Issues/PRs references
N/A

### CODE REVIEW

File: sys/net/application_layer/nanocoap/nanocoap.c [hunk 1]
```diff
@@ -436,6 +448,19 @@ void coap_pkt_init(coap_pkt_t *pkt, uint8_t *buf, size_t len, size_t header_len)
...
+ *
+ * val[in]               Value of a nybble of the first byte of the option
+ *                       header. Lower nybble is coded length of delta; upper
+ *                       nybble is coded length of value.
+ * pkt_pos_ptr[in,out]   in: commonly, first byte of the option's value;
+ *                       otherwise, first byte of extended delta/length header
+ *                       out: next byte after the value or extended header
+ * pkt_end[in]           next byte after all options
+ *
+ * return                field value
```
Comment (leandrolanzieri) [inline sys/net/application_layer/nanocoap/nanocoap.c]: You could also add the `-EBADMSG` return value in case of Payload Marker.

Comment (kb2ma) [reply sys/net/application_layer/nanocoap/nanocoap.c]: Good catch. I made another pass through the return values.

File: sys/net/application_layer/nanocoap/nanocoap.c [hunk 2]
```diff
@@ -176,6 +176,18 @@ uint8_t *coap_find_option(const coap_pkt_t *pkt, unsigned opt_num)
     return NULL;
 }
 
+/*
+ * Parse option attributes
+ *
+ * pkt[in]           coap_pkt_t for buffer
+ * pkt_pos[in,out]   in: first byte of option in buffer
```
Comment (kaspar030) [inline sys/net/application_layer/nanocoap/nanocoap.c]: how can this be "out"? It is a pointer, not pointer to pointer.

Comment (kb2ma) [reply sys/net/application_layer/nanocoap/nanocoap.c]: Thanks for correcting the sloppiness.

**Comment by kb2ma** [comment]
Squashed. Thanks @smlng and all.

**Comment by smlng** [comment]
@kaspar030 good to go here?

**Comment by kaspar030** [comment]
> @kaspar030 good to go here?

yup!

**Pull request merged**